drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py

```diff
@@ -0,0 +1,758 @@
+
+import sample_tracking_helper_qbic as tracking_helper
+#### Setup Sample Tracking service
+SERVICE_CREDENTIALS = ServiceCredentials()
+SERVICE_CREDENTIALS.user = tracking_helper.get_service_user()
+SERVICE_CREDENTIALS.password = tracking_helper.get_service_password()
+SERVICE_REGISTRY_URL = URL(tracking_helper.get_service_reg_url())
+QBIC_LOCATION = tracking_helper.get_qbic_location_json()
+
+### We need this object to update the sample location later
+SAMPLE_TRACKER = SampleTracker.createQBiCSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS, QBIC_LOCATION)
+
+######## Sample Tracking related import
+from life.qbic.sampletracking import SampleTracker
+from life.qbic.sampletracking import ServiceCredentials
+from java.net import URL
+
+import sample_tracking_helper_qbic as tracking_helper
+
+#############################################################################
+#
+# The ETL environment setup.
+#
+#############################################################################
+# Path to checksum.py
+sys.path.append('/home-link/qeana10/bin')
+
+# String template for the CentraXX XML naming specification
+CENTRAXX_XML_NAME = '{patient_id}_{sample_id}_patient_centraxx.xml'
+
+# Regex matching the QBiC barcode naming specification
+QCODE_REG = re.compile('Q\w{4}[0-9]{3}[a-zA-Z]\w')
+
+# Regex matching the RNAseq sample file naming specification
+RNASEQ_REG = re.compile(r'.*tumor_rna.[1,2]{1}.fastq.gz')
+
+# Path to the openBIS properties file
+PROPERTIES = '/etc/openbis.properties'
+
+# Some openBIS type definitions
+NGS_SAMPLE_TYPE = 'Q_NGS_SINGLE_SAMPLE_RUN'
+NGS_EXP_TYPE = 'Q_NGS_MEASUREMENT'
+NGS_RAW_DATA = 'Q_NGS_RAW_DATA'
+MTB_SAMPLE_TYPE = 'Q_NGS_MTB_DIAGNOSIS_RUN'
+MTB_EXP_TYPE = 'Q_NGS_MTB_DIAGNOSIS'
+MTB_RAW_DATA = 'Q_NGS_MTB_DATA'
+NGS_VARIANT_CALL = 'Q_NGS_VARIANT_CALLING'
+
+#### Setup Sample Tracking service
+SERVICE_CREDENTIALS = ServiceCredentials()
+SERVICE_CREDENTIALS.user = tracking_helper.get_service_user()
+SERVICE_CREDENTIALS.password = tracking_helper.get_service_password()
+SERVICE_REGISTRY_URL = URL(tracking_helper.get_service_reg_url())
+QBIC_LOCATION = tracking_helper.get_qbic_location_json()
+
+### We need this object to update the sample location later
+SAMPLE_TRACKER = SampleTracker.createQBiCSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS, QBIC_LOCATION)
+
+
+# Experiment ID counter
+EXPERIMENT_ID = 0
+
+# Check if the mtbconverter executable is found in the system path
+cmd_status = mtbutils.mtbconverter(['-h'])
+
+if cmd_status != 0:
+    raise mtbutils.MTBdropboxerror("Mtbconverter could not be loaded: " + cmd_status)
+
+print(mtbutils.log_stardate("Mtbconverter executable found."))
+
+config = ConfigParser.ConfigParser()
+config.read(PROPERTIES)
+
+api = HttpInvokerUtils.createServiceStub(IApplicationServerApi, config.get('openbis','url') + IApplicationServerApi.SERVICE_URL, 5000)
+ 
+sessionToken = api.login(config.get('openbis','user'), config.get('openbis','password'))
+
```

Comment:
Why is this needed in the jython script?

Reply:
V3 api usage